examples/QuickJournal/QuickJournal/ViewModels/JournalEntryDetailsViewModel.cs

```diff
@@ -1,36 +1,52 @@
-﻿using System.Windows.Input;
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using QuickJournal.Models;
+using QuickJournal.Services;
 using Realms;
 using Xamarin.Forms;
 
-namespace QuickJournal
+namespace QuickJournal.ViewModels
 {
-    public class JournalEntryDetailsViewModel
+    public class JournalEntryDetailsViewModel : BaseViewModel
     {
-        private Transaction _transaction;
+        private Transaction transaction;
 
         public JournalEntry Entry { get; private set; }
 
-        internal INavigation Navigation { get; set; }
-
         public ICommand SaveCommand { get; private set; }
 
         public JournalEntryDetailsViewModel(JournalEntry entry, Transaction transaction)
         {
+            this.transaction = transaction;
             Entry = entry;
-            _transaction = transaction;
-            SaveCommand = new Command(Save);
+
+            InitCommands();
+        }
+
+        private void InitCommands()
+        {
+            SaveCommand = new Command(async () => await Save());
         }
 
-        private void Save()
+        private async Task Save()
         {
-            _transaction.Commit();
-            Navigation.PopAsync(true);
+            if (string.IsNullOrEmpty(Entry.Title))
+            {
+                await DialogService.ShowAlert("Error", "You cannot save a note with an empty title");
+                return;
+            }
+
+            Entry.Metadata.LastModifiedDate = DateTimeOffset.Now;
+            transaction.Commit();
+            transaction = null;
```

Comment:
What advantages do you gain in setting the transaction to null and checking OnDisappearing if it's still around to dispose of it?

Reply:
`OnDisappearing` gets called every time the page disappears, both when the user just presses the back button and when we programmatically go back to the previous page because the user wants to save the note. 
By setting it to null after saving we avoid disposing of a transaction that was already committed. It actually works anyways (no exceptions or anything) but it seems like terrible style 😁

Reply:
Do we need to dispose it at all? In other words (not a Xamarin lifecycle expert here): Could we just avoid `null` and `Dispose` and let the garbage collection take care of it?

Reply:
I wasn't imply it was a mistake or anything. It was pure curiosity. Thank you for the explanation.

Reply:
@DominicFrei I think you gain that the GC doesn't get in that condition where it has to stop everything and do a round of cleaning, because you clean when you know it's time to. I assume this is the right way to do it in any language with automatic memory management.
@papafe am I right?

Reply:
So... Here more than the memory management I'm worrying about leaving the db in a "valid" state. When we dispose of the transaction it also rollbacks all the changes that have been done since the transaction has been opened. If we leave it to the GC, we wouldn't know when this happens. 
Looking at the source code I can also see that it's never an issue to dispose of a transaction after it was committed (or viceversa actually).

Reply:
Fair point. Choosing the safe approach is surely the right way to go. Is this also documented so that users know about it? Do we always follow this rule consistently and how can we make sure we do?

Reply:
I was just reading the new documentation, and it seems it does not say anywhere about the possibility of opening a transaction explicitly (not using `realm.Write`). It's definitely something that should be added

Reply:
I wonder if we should add `State` property to the `Transaction` type. Seems a bit silly that we're throwing exceptions if the user tries to double commit or rollback after a commit, but we don't offer an API for them to check what the state of the transaction is and if they can call methods on it. If we had a `State` property, we could then check it in `OnDisappearing` and have something like `if (transaction.State == TransactionState.Active) { rollback() }`.

Reply:
Agree, opened #2551